src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -3197,13 +3197,16 @@ def openshift_create(cmd, client, resource_group_name, name,  # pylint: disable=
                 namespace='Microsoft.Network', type='virtualNetwork',
                 name=vnet_peer
             )
-    if workspace_resource_id is not None:
-        workspace_resource_id = workspace_resource_id.strip()
-        if not workspace_resource_id.startswith('/'):
-            workspace_resource_id = '/' + workspace_resource_id
-        if workspace_resource_id.endswith('/'):
-            workspace_resource_id = workspace_resource_id.rstrip('/')
-        monitor_profile = OpenShiftManagedClusterMonitorProfile(enabled=True, workspace_resource_id=workspace_resource_id)  # pylint: disable=line-too-long
+    if workspace_id is not None:
+        workspace_id = workspace_id.strip()
+        if not workspace_id.startswith('/'):
+            workspace_id = '/' + workspace_id
+        if workspace_id.endswith('/'):
+            workspace_id = workspace_id.rstrip('/')
+        if len(workspace_id) == 36 and bool(re.match(r'^(\d|\w{8})-(\d|\w{4})-(\d|\w{4})-(\d|\w{4})-(\d|\w{12})$', workspace_id)):  # pylint: disable=line-too-long
+            subscription = get_subscription_id(cmd.cli_ctx)
+            workspace_id = '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.OperationalInsights/workspaces/{}'.format(subscription, resource_group_name, workspace_id)  # pylint: disable=line-too-long
```

Comment:
Is it possible for server end?

Reply:
@JackQuincy mentioned in the other PR that this has to be done client side.  AKS does the same validation for their workspace-resource-id on line 2204.  

The regex validation was checking the subscription but that is wrong.  Changing it to append the name of the workspace resource id to the string to build the monitor profile.

Reply:
Can you add a test for this command? It's a little bit strange to have a id like `'/subscriptions/{}/resourceGroups/{}/providers/Microsoft.OperationalInsights/workspaces/{}'.format(subscription, resource_group_name, workspace_id)` The workspace_id is something like xxxx-xxxx-xxxx-xxxx-xxxx-xxxx which is strange.

Reply:
@MyronFanQiu  I'm confused.  When you do `az resource list` the **id** of the loganalytics workspace is this string:
 ```
/subscriptions/{subscription}/resourceGroups/{resourceGroup}/providers/Microsoft.OperationalInsights/workspaces/{workspace-name}
```
The `workspace name` is the just 4-63 characters after `/workspaces/`.  The logic allows for either the string to be passed in with the `workspace name` as the `id` or just the workspace name which builds the id because the OpenshiftMonitorProfile needs the whole id to create the managed cluster. 

I will indeed create a test for the functionality of the above logic.  

The id I believe you are describing is the subscription id that is passed with the id of the loganalytics workspace id.  Unless I'm mistaken.

Reply:
@troy0820 Oh. Since the name of the argument is `workspace-id`, I thought this command supported both `workspace_resource_id` and `workspace_id`. If you are intend to support Name or Resource ID. Usually, the name of the argument should be just `workspace` and specify the detail information in the help message. We still need a test for this command. Thanks~

Reply:
I will add a test for this but will also wait until this PR https://github.com/Azure/azure-cli/pull/11460 is merged and I can rebase.  It changes the imports to not obscure the API version from the caller and only use the preview 0930 imports.